epoxy-processor/src/main/java/com/airbnb/epoxy/GeneratedModelWriter.kt

```diff
@@ -1494,28 +1498,54 @@ internal class GeneratedModelWriter(
-                beginControlFlow("if (properties.has(\$S))", setterName)
-                val jsonGetterName = when {
-                    attributeInfo.isBoolean -> "getBoolean"
-                    attributeInfo.isCharSequenceOrString
-                            || attributeInfo.isStringAttributeData -> "getString"
-                    attributeInfo.isDouble -> "getDouble"
-                    attributeInfo.isDrawableRes -> "getDrawableRes"
-                    attributeInfo.isEpoxyModelList -> "getEpoxyModelList"
-                    attributeInfo.isInt && !attributeInfo.isDrawableRes -> "getInt"
-                    attributeInfo.isStringList -> "getStringList"
-                    attributeInfo.isViewClickListener -> "getOnClickListener"
-                    else -> throw IllegalStateException("Missing ModelProperties method for a supported attribute type.")
+            // Groups attributes that are part of the same attribute group, others will be by
+            // themselves (at this point no two attributes should have the same generated setter
+            // name).
+            val supportedAttributeInfoGroups = supportedAttributeInfo.groupBy {
+                if (it.groupKey.isNullOrEmpty()) {
+                    it.generatedSetterName()
+                } else {
+                    it.groupKey
+                }
+            }
+
+            for ((_, attributeInfoGroup) in supportedAttributeInfoGroups) {
+                for ((index, attributeInfo) in attributeInfoGroup.withIndex()) {
+                    val setterName = attributeInfo.generatedSetterName()
+                    val isStartOfGroup = index == 0
+                    val isEndOfGroup = index == attributeInfoGroup.size - 1
+
+                    if (isStartOfGroup) {
+                        beginControlFlow("if (properties.has(\$S))", setterName)
+                    } else {
+                        nextControlFlow("else if (properties.has(\$S))", setterName)
+                    }
+
+                    val jsonGetterName = when {
+                        attributeInfo.isBoolean -> "getBoolean"
+                        attributeInfo.isCharSequenceOrString
+                                || attributeInfo.isStringAttributeData -> "getString"
+                        attributeInfo.isDouble -> "getDouble"
+                        attributeInfo.isDrawableRes -> "getDrawableRes"
+                        attributeInfo.isEpoxyModelList -> "getEpoxyModelList"
+                        attributeInfo.isInt && !attributeInfo.isDrawableRes && !attributeInfo.isRawRes -> "getInt"
+                        attributeInfo.isLong -> "getLong"
+                        attributeInfo.isRawRes -> "getRawRes"
+                        attributeInfo.isStringList -> "getStringList"
+                        attributeInfo.isViewClickListener -> "getOnClickListener"
+                        else -> throw IllegalStateException("Missing ModelProperties method for a supported attribute type.")
```

Comment:
should use the errorLogger instead of throwing, so generating the rest of the models is not blocked. Otherwise error output is terrible since models don't exist